### PR TITLE
Let nessie-quarkus use resteasy-reactive

### PR DESCRIPTION
**Pull request by snazy**
* Replace `io.quarkus:quarkus-resteasy-*` with `io.quarkus:quarkus-resteasy-reactive-*`
* Adopt exception mappers to the different behavior of Jackson exceptions
* Cleanup: remove left-over Graal/native files
* Cleanup: eliminate `private` injected field warning in `BaseExceptionMapper`

### CODE REVIEW

File: servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ * Type, Annotation[], MediaType, MultivaluedMap, InputStream)} throws a {@link
+ * WebApplicationException} for {@link DatabindException} ({@link StreamReadException}), which makes
+ * it impossible to install an exception mapper for those, so we have to install one for {@link
+ * WebApplicationException}.
+ *
+ * <p>See also {@link MismatchedInputExceptionMapper} for a special case of a "bad request".
+ */
+@Provider
+@ApplicationScoped
+public class WebApplicationExceptionRemapper extends BaseExceptionMapper<WebApplicationException> {
```
Comment (adutra) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java:49]: Nit: why "remapper" and not "mapper"?

File: servers/quarkus-server/src/main/java/org/projectnessie/server/rest/MismatchedInputExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * "Special" implementation for {@link MismatchedInputException}s that represent bad input
+ * parameters.
+ *
+ * <p>See also {@link WebApplicationExceptionRemapper} for a special case of a "bad request".
+ */
+@Provider
+@ApplicationScoped
+public class MismatchedInputExceptionMapper extends BaseExceptionMapper<MismatchedInputException> {
```
Comment (adutra) [inline servers/quarkus-server/src/main/java/org/projectnessie/server/rest/MismatchedInputExceptionMapper.java:33]: We are already mapping `JsonMappingException` which is a super class, in `NessieJaxRsJsonMappingExceptionMapper`. Why do we need the map this specific subtype as well?

File: servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java (hunk 1 above)
Comment (adutra) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java:49]: Also, just out of curiosity, why are some mappers declared in nessie-quarkus, and others in nessie-rest-common?

**Review by adutra — APPROVED** [review]

File: servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java (hunk 1 above)
Comment (snazy) [reply servers/quarkus-server/src/main/java/org/projectnessie/server/rest/WebApplicationExceptionRemapper.java:49]: I _think_ `WebApplicationException` already (kind of) mapped elsewhere.

> Also, just out of curiosity, why are some mappers declared in nessie-quarkus, and others in nessie-rest-common?

The Quarkus-specific ones are in `:nessie-quarkus`

**Pull request merged**